### ISSUE TITLE
Logfile grows to 9gb and appears to be caused by altitude-related loop in space battle

### ISSUE THREAD

**Issue by Duuvian** [MegaMek/megamek#4370]
### Environment
MegaMek v0.49.13-SNAPSHOT
	Build Date: 2023-04-24T01:20:35.088117800
	Today: 2023-04-25
	Origin Project: MekHQ
	Java Vendor: Eclipse Adoptium
	Java Version: 11.0.16.1
	Platform: Windows 10 10.0 (amd64)
	System Locale: en_US
	Total memory available to MegaMek: 2 GB

### Description
I mistakenly posted this on a pull request instead of leaving a comment on Issue #3696 that the pull request fixed. The pull request seemed related to this new issue to some degree, but the posted comment on the pull request didn't appear on Insights so I made a new issue, I apologize if that was in error or if it actually is unrelated.

Note taken at time of bug:
Bots in a space battle "breakthrough" where an enemy dropship has to make it to the other side did not do anything and it did not advance to other units' turn to move. Kicking the bot resulted in a temporary freeze, and the next unit also did not move.

When I went to gather the logs I noticed the megamek log was just under 9 gigabytes large. I did not attach this log for this reason. On a third attempt to see if the logsize could be lowered, I closed megamek after about 20 seconds and found the attached logs. The log suggests a bug with receiving elevation for aerospace in a space map that causes a loop or something. I'll attach this smaller log file if it fits onto github though it is still massive for a text file. (Note: it does not, it's limited to 25mb and 756 MB was attempted to upload). Here is a screenshot of the 9gb logfile:

![9gb log file](https://user-images.githubusercontent.com/84265834/235130324-9e0474fc-5590-428d-aba4-07721f2939ca.png)

I attached the save file, it has no customs. To replicate the circumstances, assign all the player units in the scenario to the integrated allies bot before starting the match. The map is huge so you may want to lower that to 50x50 or something; I did this on my second attempt as I thought it could have been caused by too large of a map but the bug still occurred.

Here is a copy and paste of what appears to be looping in the log file:

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

16:49:57,798 WARN  [megamek.common.Entity] {Princess-precognition (Magistracy of Canopus Allied Force)}
megamek.common.Entity.getAltitude(Entity.java:13142) - Altitude retrieved for a spaceborne unit!

### Files
No customs
[Pleasurebots30230425.cpnx.gz](https://github.com/MegaMek/megamek/files/11352775/Pleasurebots30230425.cpnx.gz)
